TypeSystem/flow-analysis/promotion_via_type_test_A04_t06.dart

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We say that a variable `x` is promotable via type test with type
+/// `T` given variable model `VM` if
+/// - `VM = VariableModel(declared, promoted, tested, assigned, unassigned,
+///     captured)`
+/// - and `captured` is `false`
+/// - and `S` is the current type of `x` in `VM`
+/// - and not `S <: T`
+/// - and `T <: S` or (`S` is `X extends R` and `T <: R`) or (`S` is `X & R` and
+///   `T <: R`)
+///
+/// @description Checks that a variable is promoted after the type test in one
+/// of the branches, then is is not promoted in another branch and after it.
+/// Test type `Never`.
+/// @author sgrekhov22@gmail.com
+
+import '../../Utils/static_type_helper.dart';
+
+test1() {
+  String? s = 1 > 2 ? null : "a";
+  if (s is Never) {
+  } else {
```

Comment:
We do have another test (or several) about the outcome of testing for `Never`, but it might still be good for the overall consistency and readability of this test to add this to the "then" part:

  s.expectStaticType<Exactly<String?>>(); // Dead code and no promotion.

Reply:
Thanks!